src/shims/x86/sse2.rs

```diff
@@ -243,22 +238,22 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                         << 32)
                     | (u64::from(this.read_scalar(&this.project_index(&right, 3)?)?.to_u16()?)
                         << 48);
+                // Convert it to the type expected by checked_{shl,shr}
+                let shift = shift.try_into().unwrap_or(u32::MAX);
```

Comment:
Why is it okay to fall back to u32::MAX here? Please add a comment.
Do we have a test covering this case?

Reply:
Tests could definitely be improved. I probably will contribute them also to stdarch.